lib/pbkdf2.dart

```diff
@@ -13,18 +13,18 @@ class PBKDF2 {
   /// Creates instance capable of generating a key.
   ///
   /// [hashAlgorithm] defaults to [sha256].
-  PBKDF2({Hash hashAlgorithm}) {
+  PBKDF2({Hash? hashAlgorithm}) {
     this.hashAlgorithm = hashAlgorithm ?? sha256;
   }
 
-  Hash get hashAlgorithm => _hashAlgorithm;
-  set hashAlgorithm(Hash algorithm) {
+  Hash? get hashAlgorithm => _hashAlgorithm;
+  set hashAlgorithm(Hash? algorithm) {
     _hashAlgorithm = algorithm;
-    _blockSize = _hashAlgorithm.convert([1, 2, 3]).bytes.length;
+    _blockSize = _hashAlgorithm!.convert([1, 2, 3]).bytes.length;
   }
 
-  Hash _hashAlgorithm;
-  int _blockSize;
+  Hash? _hashAlgorithm;
+  late int _blockSize;
```

Comment:
I would think that since _blockSize depends on _hashAlgorithm, _blockSize would need to be nullable.